healthyregions/SDOHPlace
# Similar results short-term solutions

This PR represents the front-end implementation of the 'Similar Result' section (#313). I will create a separate PR for the metadata manager to update the Solr configuration.

After multiple attempts, I found that the pre-filter for the 'suppressed' attribute in the `suggest` endpoint still not working. To optimize it as much as possible for now, I've implemented front-end adjustment: Once all of the suggestions (max 200) have been returned, the autocomplete will filter out terms with titles that include the '(year)' format, such as 'XXXIndex (2012),' since these are likely child records. 

This solution is not perfect, as some records may still appear, but we can improve it in the future.

Another challenge is the multiple asynchronous calls triggered by autocomplete suggestion calls, term search calls, and other searches that match each suggestion of the final term. These caused overwritten results and delays in presenting them. I updated the handleSearch mechanism to optimize this.

### How to Test

1. Navigate to the search page and type 'a'.  ~~You should now see all location-like terms have been moved to the bottom of the suggestion dropdown~~ (In my latest Solr configuration, I managed to remove location terms related to spatial coverage (such as "Illinois") from the Solr side, **so you should not see any spatial coverage data appear in the auto-complete now**) :
<img width="1297" alt="Screenshot 2024-10-28 at 11 47 03 AM" src="https://github.com/user-attachments/assets/724a283c-a4c0-4234-877b-96e863c4af6f">

2. Press Enter to simulate the audience searching for 'a', which is a case expected to generate a large amount of related results. You should see a progress bar appear for a few seconds, followed by the recommended results appearing:

<img width="790" alt="Screenshot 2024-10-28 at 11 47 54 AM" src="https://github.com/user-attachments/assets/e0050d9d-452d-4037-a291-dd5448409ee6">

Note: I changed "Similar Results" to "You may want to check," as we're now adopting a more recommendation-style approach.

3. Choose the "average download speed" option from the dropdown. You should receive only one result since there is no additional suggestion for this query:

<img width="1645" alt="Screenshot 2024-10-28 at 11 50 30 AM" src="https://github.com/user-attachments/assets/fc094223-c7c4-48b8-a757-627817e441a6">

4. Search for "soc" without selecting anything from the dropdown. You should receive the "You may want to check" results faster since there are fewer suggestions to search for. **Using a higher weight could speed this up more (currently I use 50/100). Let me know if you want to use a higher bar**:

<img width="1246" alt="Screenshot 2024-10-28 at 11 52 45 AM" src="https://github.com/user-attachments/assets/c1ae6b00-7e74-418a-b07e-aaf7814790b9">

5. You should be able to review the details of a suggested item by clicking the "View" button on any object.

<img width="1720" alt="Screenshot 2024-10-28 at 12 14 25 PM" src="https://github.com/user-attachments/assets/40ca2ad4-035f-49e5-94f9-1c11ce46abc6">
